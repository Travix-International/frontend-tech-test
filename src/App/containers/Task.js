import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Route } from 'react-router-dom';

import TaskPage from  '../components/TaskPage';
import Form from '../components/Form';
import * as actionCreators from '../actions/task';

class Task extends React.Component {
  componentDidMount() {
    this.props.actions.getTask(this.props.match.params.id);
  }

  render() {
    const { match, actions, task, location, history } = this.props;
    const editable = location.pathname.indexOf('edit') !== -1;

    return (
      <div className="container" >
        {
          editable ?
          (
            <TaskPage
              editable
              update={actions.update.bind(null, match.params.id)}
              task={task}
              history={history}
            />
          )
          :
          (
            <TaskPage
              task={task}
            />
          )
        }
      </div>
    )
  }
}

function mapStateToProps(state) {
  return { task: state.task }
}


function mapDispatchToProps(dispatch) {
  return { actions: bindActionCreators(actionCreators, dispatch) }
}

export default connect(mapStateToProps, mapDispatchToProps)(Task)