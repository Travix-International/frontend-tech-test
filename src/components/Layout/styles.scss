.layout {
  background-image: url(./bg.jpg);
  font-family: 'Montserrat', sans-serif;
  min-height: 768px;
}

.layout-container {
  max-width: 1200px;
  margin: 0 auto;
  font-size: 16px;
  padding-top: 100px;
}