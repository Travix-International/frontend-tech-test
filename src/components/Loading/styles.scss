@-moz-keyframes spin {
    from { -moz-transform: rotate(0deg); }
    to { -moz-transform: rotate(360deg); }
}
@-webkit-keyframes spin {
    from { -webkit-transform: rotate(0deg); }
    to { -webkit-transform: rotate(360deg); }
}
@keyframes spin {
    from {transform:rotate(0deg);}
    to {transform:rotate(360deg);}
}

.loading {
    position: fixed;
    top: 0px;
    left: 0px;
    bottom: 0px;
    right: 0px;
    z-index: 1000;
    background-color: rgba($color: #000000, $alpha: 0.5);
    display: none;

    .fa-spinner {
        position: absolute;
        top: 50%;
        left: 50%;
        color: white;
        font-size: 4em;
        animation: spin 1s infinite linear;
    }

    &.show {
        display: block;
    } 
}