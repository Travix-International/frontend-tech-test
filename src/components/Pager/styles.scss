.pager {
    position: fixed;
    bottom: 0px;
    left: 0px;
    right: 0px;
    z-index: 100;
    background-color: white;

    .pagination {
        padding: 0px;
        margin: 0px auto;
        max-width: 1200px;
        line-height: 30px;
        list-style: none;
        text-align: center;

        li {
            float: left;
            padding: 0px 5px;
            margin: 0px 3px;
            width: 29px;

            &.disabled {
                cursor: not-allowed;
            }

            &.active {
                border: 1px solid black;
            }

            a {
                text-decoration: none;
            }
        }
    }
}