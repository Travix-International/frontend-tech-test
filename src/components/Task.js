const React = require('react');

const Task = ({ title, description }) => (
  <li>
    <p>title: {title}</p>
    <p>description: {description}</p>
  </li>
);

module.exports = Task;