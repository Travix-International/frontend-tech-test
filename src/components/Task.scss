@import "./ToggleButton.scss";
@import "./DeleteButton.scss";

.task {
  display: block;
  width: 100%;
  max-width: 500px;
  border-radius: .5rem;
  background-color: #fff;
  border: none;
  margin: .5rem auto;
  font-size: 1rem;
  font-weight: 300;
  color: #888;
  padding: 1rem 3rem;
  position: relative;
  transition: background-color 100ms linear;
  
  &--completed {
    text-decoration: line-through;
    background-color: transparent;
    background-image: repeating-linear-gradient(45deg, transparent, transparent .5rem, #fff .5rem, #fff 1rem);
    color: lighten(#888, 20%);
    
    .task__title {
      color: lighten(#888, 20%);
    }
  }
      
  &__title {
    font-family: 'Varela Round';
    color: #0087cd;
    font-size: 1.5rem;
  }
  
  &__description {
    margin-bottom: 0;
  }
}

@media (min-width: 1024px) {
  .task {
    max-width: none;
  }
}