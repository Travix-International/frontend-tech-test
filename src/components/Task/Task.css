.icon-edit {
  fill: #999; }
