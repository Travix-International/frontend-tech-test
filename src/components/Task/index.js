const React = require('react');
const { connect } = require('react-redux');
const { updateTask } = require('../../actions');
const ToggleButton = require('../ToggleButton');
const DeleteButton = require('../DeleteButton');
const PropTypes = require('prop-types');

class Task extends React.Component {
  constructor() {
    super();
    this.titleElement = React.createRef();
    this.descriptionElement = React.createRef();
    this.update = this.update.bind(this);
  }

  update() {
    this.props.dispatch(updateTask({
      id: this.props.id,
      title: this.titleElement.current.textContent.trim(),
      description: this.descriptionElement.current.textContent.trim(),
    }))
  }

  render() {
    const { id, title, description, completed } = this.props;
    return (
      <li className={`card task${completed ? ' task--completed' : ''}`}>
        <header
          className="card__title"
          contentEditable={!completed}
          onBlur={this.update}
          ref={this.titleElement}
          suppressContentEditableWarning={!completed}
        >{title}</header>
        <p
          className="card__description"
          contentEditable={!completed}
          onBlur={this.update}
          ref={this.descriptionElement}
          suppressContentEditableWarning={!completed}
        >{description}</p>

        <ToggleButton completed={completed} id={id} />
        <DeleteButton id={id} />
      </li>
    );
  };
};

Task.propTypes = {
  completed: PropTypes.bool,
  description: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
};

Task.defaultProps = {
  completed: false,
};

module.exports = connect()(Task);
