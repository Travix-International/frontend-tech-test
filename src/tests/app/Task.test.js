import React from 'react';
import {Icon, Button} from 'antd'
import { shallow } from 'enzyme';
import {taskWithIsDeletedFalse, taskWithIsDeletedTrue} from './test.stub';
import Task from '../../containers/app/Task';
import styles from '../../containers/app/app.module.scss';

describe('Task Component', () => {
    it('should render', () => {
      const component = shallow(<Task task={taskWithIsDeletedFalse} />);
      expect(component).toBeTruthy();
    });
    it('should render Task Icon', () => {
        const component = shallow(<Task task={taskWithIsDeletedFalse} />);
        expect(component.find(Icon).length).toBe(1);
    });
    it('should render Task Title', () => {
        const component = shallow(<Task task={taskWithIsDeletedFalse} />);
        expect(component.find('span#title').length).toBe(1);
    });

    it('should render Task Description', () => {
        const component = shallow(<Task task={taskWithIsDeletedFalse} />);
        expect(component.find('span#description').length).toBe(1);
    });

    it('should render Edit and Delete Task Icon', () => {
        const component = shallow(<Task task={taskWithIsDeletedFalse} />);
        expect(component.find(Button).length).toBe(2)
    });
    it('should render Done Task Icon', () => {
        const component = shallow(<Task task={taskWithIsDeletedTrue} />);
        expect(component.find(Button).length).toBe(1);
    });

    it('should render Task when task is taskWithIsDeletedTrue', () => {
        const component = shallow(<Task task={taskWithIsDeletedTrue} />);
        expect(component.find('strike').length).toBe(1);
    });
});