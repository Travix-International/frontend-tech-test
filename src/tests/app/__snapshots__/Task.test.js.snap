// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render TaskList Component 1`] = `<Connect(TasksList) />`;

exports[`App Component should render correctly in "debug" mode 1`] = `
<div>
  <nav
    className="nav"
  >
    <div
      className="header"
    >
      o Do Lis
    </div>
  </nav>
  <Connect(TasksList) />
</div>
`;

exports[`Task Component should render 1`] = `
<div
  className="task"
>
  <div>
    <Icon
      type="bars"
    />
    <span
      className="title"
    >
       
       
    </span>
    <span
      className="description"
    />
  </div>
  <div
    className="taskButtonHolder"
  >
    <Button
      block={false}
      ghost={false}
      icon="edit"
      loading={false}
      onClick={[Function]}
      type="default"
    />
    <Button
      block={false}
      ghost={false}
      icon="delete"
      loading={false}
      onClick={[Function]}
      type="primary"
    />
  </div>
</div>
`;

exports[`Task Component should render 2`] = `
<div
  className="task"
>
  <div>
    <Icon
      type="bars"
    />
    <span
      className="title"
    >
       
       
    </span>
    <span
      className="description"
    />
  </div>
  <div
    className="taskButtonHolder"
  >
    <Button
      block={false}
      ghost={false}
      icon="edit"
      loading={false}
      onClick={[Function]}
      type="default"
    />
    <Button
      block={false}
      ghost={false}
      icon="delete"
      loading={false}
      onClick={[Function]}
      type="primary"
    />
  </div>
</div>
`;

exports[`Task Component should render Task when task is taskWithIsDeletedFalse 1`] = `
<div
  className="task"
>
  <div>
    <Icon
      type="bars"
    />
    <span
      className="title"
    >
       
      Vaughan
       
    </span>
    <span
      className="description"
    >
      minim
    </span>
  </div>
  <div
    className="taskButtonHolder"
  >
    <Button
      block={false}
      ghost={false}
      icon="edit"
      loading={false}
      onClick={[Function]}
      type="default"
    />
    <Button
      block={false}
      ghost={false}
      icon="delete"
      loading={false}
      onClick={[Function]}
      type="primary"
    />
  </div>
</div>
`;

exports[`Task Component should render Task when task is taskWithIsDeletedTrue 1`] = `
<div
  className="task"
>
  <div>
    <Icon
      type="bars"
    />
    <span
      className="title"
    >
      <strike>
         
        Vaughan
          
      </strike>
    </span>
  </div>
  <div
    className="taskButtonHolder"
  >
    <Button
      block={false}
      disabled={true}
      ghost={false}
      icon="check"
      loading={false}
      type="default"
    />
  </div>
</div>
`;

exports[`Task Component should render correctly in "debug" mode 1`] = `
<div
  className="task"
>
  <div>
    <Icon
      type="bars"
    />
    <span
      className="title"
    >
       
       
    </span>
    <span
      className="description"
    />
  </div>
  <div
    className="taskButtonHolder"
  >
    <Button
      block={false}
      ghost={false}
      icon="edit"
      loading={false}
      onClick={[Function]}
      type="default"
    />
    <Button
      block={false}
      ghost={false}
      icon="delete"
      loading={false}
      onClick={[Function]}
      type="primary"
    />
  </div>
</div>
`;

exports[`Task Component should render not deleted Task 1`] = `
<div
  className="task"
>
  <div>
    <Icon
      type="bars"
    />
    <span
      className="title"
    >
       
      Vaughan
       
    </span>
    <span
      className="description"
    >
      minim
    </span>
  </div>
  <div
    className="taskButtonHolder"
  >
    <Button
      block={false}
      ghost={false}
      icon="edit"
      loading={false}
      onClick={[Function]}
      type="default"
    />
    <Button
      block={false}
      ghost={false}
      icon="delete"
      loading={false}
      onClick={[Function]}
      type="primary"
    />
  </div>
</div>
`;
